pysidt/sidt.py

```diff
@@ -170,6 +172,7 @@ def generate_tree(self,data=None,check_data=True):
         """
         generate nodes for the tree based on the supplied data
         """
+        self.skip_nodes = []
```

Comment:
It makes more sense to use `set` for skip_notes to keep the search time to `O(1)`